soc/arm/nuvoton_numaker/Kconfig.defconfig

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2023 Nuvoton Technology Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+source "soc/arm/nuvoton_numaker/*/Kconfig.defconfig.series"
+
+config RESET
+	default y
```

Comment:
there's no reset driver at this point, please, add stuff in a bisectable way. in this case, this needs to be added by the introduction of the reset driver.

Reply:
So, should I move this update of Kconfig.defconfig in the commit of reset driver ?

Reply:
yes please, also review for any other similar issues.

Reply:
Thanks, I also move the clock control update of soc.c to the commit of clock-control driver.
Fixed above issues and force-pushed commits.

Reply:
Hi @gmarull,
Fixed and please have a look.

Reply:
Hi @gmarull,
I updated the code based on your requested changes.
Please have a look.

Reply:
Hi @gmarull,
Finished your requested changes.
Please revisit.